src/Build/BackEnd/Components/RequestBuilder/IntrinsicTasks/ItemGroupLoggingHelper.cs

```diff
@@ -18,9 +18,14 @@ namespace Microsoft.Build.BackEnd
     internal static class ItemGroupLoggingHelper
     {
         /// <summary>
-        /// The default character limit for logging parameters.
+        /// The default character limit for logging parameters. 10k is somewhat arbitrary, see https://github.com/microsoft/msbuild/issues/4907.
+        /// </summary>
+        internal static int parameterCharacterLimit = 10_000;
+
+        /// <summary>
+        /// The default parameter limit for logging. 200 is somewhat arbitrary, see https://github.com/microsoft/msbuild/pull/5210.
         /// </summary>
-        internal static int parameterTextLimit = 10000;
+        internal static int parameterLimit = 200;
```

Comment:
Rainer said 10k or perhaps even 20k is more reasonable here.

Reply:
No, I didn't. I think it's good for the _character_ limit.

Reply:
"We've seen 10k items in C++ builds" was very confusing, then.